src/lints/declarative_macro_missing.ron

```diff
@@ -0,0 +1,46 @@
+SemverQuery(
+    id: "declarative_macro_missing",
+    human_readable_name: "macro_rules declaration removed or renamed",
+    description: "A declarative macro marked with #[macro_export] can no longer imported by its prior name.",
+    required_update: Major,
+    lint_level: Deny,
+    reference_link: Some("https://doc.rust-lang.org/reference/macros-by-example.html#path-based-scope"),
+    query: r#"
+    {
+        CrateDiff {
+            baseline {
+                item {
+                    ... on Macro {
+                        visibility_limit @filter(op: "=", value: ["$public"])
+                        public_api_eligible @filter(op: "=", value: ["$true"])
+                        name @output @tag
+
+                        span_: span @optional {
+                            filename @output
+                            begin_line @output
+                        }
+                    }
+                }
+            }
+            current @fold @transform(op: "count") @filter(op: "=", value: ["$zero"]) {
+                item {
+                    ... on Macro {
+                        visibility_limit @filter(op: "=", value: ["$public"])
+                        public_api_eligible @filter(op: "=", value: ["$true"])
```

Comment:
Our lint is for "macro removed" so we don't actually want to check `public_api_eligible` here. If we check that here, then we'd also trigger this lint if the macro became `#[doc(hidden)]`, which isn't the same lint anymore.
```suggestion
```